src/root/POSRootChainManager.ts

```diff
@@ -0,0 +1,191 @@
+import BN from 'bn.js'
+import assert from 'assert'
+import ethUtils from 'ethereumjs-util'
+import Contract from 'web3/eth/contract'
+
+import RootChainManagerArtifact from 'matic-pos-portal/artifacts/RootChainManager.json'
+import ChildTokenArtifact from 'matic-pos-portal/artifacts/ChildToken.json'
+
+import Proofs from 'matic-protocol/contracts-core/helpers/proofs.js'
+
+import ContractsBase from '../common/ContractsBase'
+import { address, SendOptions } from '../types/Common'
+import Web3Client from '../common/Web3Client'
+import RootChain from './RootChain'
+
+const logger = {
+  info: require('debug')('maticjs:POSRootChainManager'),
+  debug: require('debug')('maticjs:debug:POSRootChainManager'),
+}
+
+export default class POSRootChainManager extends ContractsBase {
+  public posRootChainManager: Contract
+  public rootChain: RootChain
+  private TRANSFER_EVENT_SIG = '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef'
+
+  constructor(posRootChainManager: address, rootChain: RootChain, web3Client: Web3Client) {
+    super(web3Client)
+    this.posRootChainManager = new this.web3Client.parentWeb3.eth.Contract(
+      RootChainManagerArtifact.abi,
+      posRootChainManager
+    )
+    this.rootChain = rootChain
+  }
+
+  async approveERC20(rootToken: address, amount: BN | string, options?: SendOptions) {
+    const txObject = this.getERC20TokenContract(rootToken, true).methods.approve(
+      this.posRootChainManager.options.address,
+      this.encode(amount)
+    )
+
+    const _options = await this._fillOptions(options, txObject, this.web3Client.getParentWeb3())
+
+    if (options.encodeAbi) {
+      _options.data = txObject.encodeABI()
+      _options.to = rootToken
+      return _options
+    }
+
+    return this.web3Client.send(txObject, _options)
```

Comment:
This chunk is duplicated too many times, can we put this in a common function? - This is also the case for methods in default Matic client, could file an issue to fix that later.

Reply:
Filing an issue for this.